app/main.py

```diff
@@ -1,3 +1,20 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Any
+import functools
+
+
+def cache(func: Callable) -> Callable:
+    """Keep a cache of previous function calls"""
+
+    @functools.wraps(func)
+    def wrapper_cache(*args: Any, **kwargs: Any) -> Any:
```

Comment:
incorrect types
args is a tuple
kwargs is a dict

Reply:
> incorrect types args is a tuple kwargs is a dict

Fixed